### PR TITLE
[5532019][AutoCast] Update GraphSanitizer to convert Double to Float32

### PR DESCRIPTION
**Pull request by ajrasane**
## What does this PR do?

**Type of change:** 
New Feature

**Overview:** 
- Implemented `convert_fp64_to_fp32()` in graph sanitizer.
- This method will be called with `GraphSanitizer.sanitize()` or `convert_to_f16()`.

## Testing
```
python -m modelopt.onnx.quantization --onnx_path=far3d_opset17.onnx  --calibration_eps cuda:0
```
- I/O sanity check passes for this model
- All unit tests pass

## Summary by CodeRabbit

* **Documentation**
  * Updated user instructions to include environment variable setup for CUDA/cuDNN libraries when using the TensorRT Docker image.

* **New Features**
  * Added automatic conversion of 64-bit floating point (FP64) tensors to 32-bit (FP32) in ONNX model graphs, enhancing compatibility and standardization during processing.

* **Tests**
  * Introduced new tests to validate correct conversion of FP64 values to FP32 across various graph components and scenarios.

### CODE REVIEW

File: modelopt/onnx/autocast/precisionconverter.py [hunk 1]
```diff
@@ -210,6 +213,32 @@ def _ensure_types_are_defined(self):
             if vi.type.tensor_type.elem_type == onnx.TensorProto.UNDEFINED:
                 vi.type.tensor_type.elem_type = self.low_precision_type.onnx_type
 
+    def _restore_original_io_types(self):
```
Comment (i-riyad) [inline modelopt/onnx/autocast/precisionconverter.py]: This is equivalent

```bash
def _restore_original_io_types(self):
    """Restore original I/O types for inputs and outputs."""
    for tensor in list(self.model.graph.input) + list(self.model.graph.output):
        original_type = self.original_network_io.get(tensor.name)
        if original_type is not None and tensor.type.tensor_type.elem_type != original_type:
            tensor.type.tensor_type.elem_type = original_type
            if tensor.name in self.value_info_map:
                self.value_info_map[tensor.name].type.tensor_type.elem_type = original_type
```

File: modelopt/onnx/autocast/precisionconverter.py [hunk 2]
```diff
@@ -210,6 +213,22 @@ def _ensure_types_are_defined(self):
...
+        """Restore original I/O types."""
+
+        def restore_tensor_type(tensor):
+            if tensor.name in self.original_network_io:
+                original_type = self.original_network_io[tensor.name]
+                if tensor.type.tensor_type.elem_type != original_type:
+                    tensor.type.tensor_type.elem_type = original_type
```
Comment (galagam) [inline modelopt/onnx/autocast/precisionconverter.py]: You're just updating the tensor's metadata here. Shouldn't we also inject a cast here?

Looking at the bigger picture, `keep_io_types` flag was covered in the unit tests, and is also set as default in MOQ, so generally we know it is working, but there's some edge case we're first seeing in https://nvbugspro.nvidia.com/bug/5532019. 
Inputs are clearly handled in lines 145-148, and outputs are presumably handled in `_cleanup_pre_output_same_type_cast`.

File: examples/onnx_ptq/README.md [hunk 3]
```diff
@@ -26,6 +26,13 @@ Model Optimizer enables highly performant quantization formats including NVFP4,
 
 Please use the TensorRT docker image (e.g., `nvcr.io/nvidia/tensorrt:25.08-py3`) or visit our [installation docs](https://nvidia.github.io/TensorRT-Model-Optimizer/getting_started/2_installation.html) for more information.
 
+Set the following environment variables inside the TensorRT docker.
+
+```bash
+export CUDNN_LIB_DIR=/usr/lib/x86_64-linux-gnu/
+export LD_LIBRARY_PATH="${CUDNN_LIB_DIR}:${LD_LIBRARY_PATH}"
+```
```
Comment (kevalmorabia97) [inline examples/onnx_ptq/README.md:34]: This is what we do in the CICD: https://nvidia.github.io/TensorRT-Model-Optimizer/getting_started/_installation_for_Linux.html#environment-setup

If that is sufficient, we can also just link this doc here. Else we can update that file too

**Review by gcunhase — APPROVED** [review]
LGTM, tested with 2 CLIs:
```sh
$ python -m modelopt.onnx.quantization --onnx_path=far3d_opset17.onnx  --calibration_eps trt cuda:0 cpu
$ python -m modelopt.onnx.quantization --onnx_path=far3d_opset17.onnx  --calibration_eps trt cuda:0 cpu --direct_io_types
```

File: examples/onnx_ptq/README.md (hunk 3 above)
Comment (ajrasane) [reply examples/onnx_ptq/README.md:34]: The steps mentioned above are a bit different for the TRT container. 
There is no `/usr/local/tensorrt/` in the TRT container
`trtexec` is stored under `/opt/tensorrt/bin`, which is already in `PATH`
And adding `/usr/lib/x86_64-linux-gnu` is sufficient for getting libcudnn.so, which is required for the `TRTExecutionProvider`.
I would suggest we keep this as it is as it will be more convenient for the user to run these steps as part of the setup right after they runs their container.
The steps you shared above should also work fine for the CI, but they are not required for the TRT container.

File: modelopt/onnx/autocast/precisionconverter.py (hunk 2 above)
Comment (ajrasane) [reply modelopt/onnx/autocast/precisionconverter.py]: I have removed this function and created conversion logic in the `GraphSanitizer`.
